Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -565,6 +570,49 @@ void FieldProbe::ComputeDiags (int step)
                 /* m_data now contains up-to-date values for:
                  *  [x, y, z, Ex, Ey, Ez, Bx, By, Bz, and S] */
                 }
+                if (do_moving_window_FP)
+                {
+                    if (temp_field_probe_integrate)
+                    {
+                        for (auto ip=0; ip < np; ip++)
+                        {
+                            amrex::ParticleReal xp, yp, zp;
+                            getPosition(ip, xp, yp, zp);
+                            if (warpx.moving_window_dir == 0)
+                            {
+                                setPosition(ip, xp+(dt*warpx.moving_window_v*PhysConst::c), yp, zp);
+                            }
+                            if (warpx.moving_window_dir == 1)
+                            {
+                                setPosition(ip, xp, yp+(dt*warpx.moving_window_v*PhysConst::c), zp);
+                            }
+                            if (warpx.moving_window_dir == WARPX_ZINDEX)
+                            {
+                                setPosition(ip, xp, yp, zp+(dt*warpx.moving_window_v*PhysConst::c));
+                            }
+                        }
+                    }
+                    else
+                    {
+                        for (auto ip=0; ip < np; ip++)
+                        {
+                            amrex::ParticleReal xp, yp, zp;
+                            getPosition(ip, xp, yp, zp);
+                            if (warpx.moving_window_dir == 0)
+                            {
+                                setPosition(ip, xp+(dt*warpx.moving_window_v*PhysConst::c)*(step-m_last_compute_step), yp, zp);
```

Comment:
Let's store `(dt*warpx.moving_window_v*PhysConst::c)*(step-m_last_compute_step)` once as constant and outside the hot loops.